### PR TITLE
[DOCS] Simplify single-node Docker instructions

### LINKED ISSUE #95791: [DOCS] Improve Install Elasticsearch with Docker documentation
There have been a few complaints about the [Install Elasticsearch with Docker](https://github.com/elastic/elasticsearch/blob/main/docs/reference/setup/install/docker.asciidoc) documentation. We should review those docs to ensure the guidance is still correct and follows best practices.

Specific feedback provided via the feedback form:

- breaks enrollment ERROR: [xpack.security.enrollment.enabled] must be set to `true` to create an enrollment token (Apparently regarding [Always bind data volumes](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html#_always_bind_data_volumes))
- step 5 doesnt work for me it is not possible to enter any key for a password
- The example does not work, as it did not create a user at the setup service, it tries to change a password for non existing user.
- Version 7 crashes on start on windows and linux, same for version 8. 
- setup fails with security_exception: unable to authenticate user [kibana_system] for REST request

Additional issues:

- [JVM heap error](https://github.com/elastic/elasticsearch/pull/96054)

### PR DESCRIPTION
**Pull request by jrodewig**
### Problem

The current [single-node Docker instructions](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html#docker-cli-run-dev-mode) are overly verbose and contain a lot of unneeded info about security internals. 

### Solution

- Restructure the above docs to focus primarily on actionable steps.
- Test the docs across operating systems (Mac, Linux, Windows) to ensure they work.

There is room for further improvement here, but I want to keep the diff small. More PRs to come.

### Issues

Closes https://github.com/elastic/elasticsearch/issues/95791
Rel: https://github.com/elastic/platform-docs-team/issues/183

### CODE REVIEW

**Review by mark-vieira — APPROVED** [review]
Thanks, James. This looks much better.

File: docs/reference/setup/install/docker.asciidoc [hunk 1]
```diff
@@ -140,78 +126,77 @@ endif::[]
...
 // NOTCONSOLE
 
-==== Enroll additional nodes
-
-When you start {es} for the first time, the installation process configures a single-node cluster by default. This process also generates an enrollment token
-and prints it to your terminal. If you want a node to join an existing cluster,
-start the new node with the generated enrollment token.
+[TIP]
+====
+To make a `curl` API call without verifying the SSL certificate, use the `-k` flag.
```
Comment (TimBosman) [inline docs/reference/setup/install/docker.asciidoc:176]: I found out that the -k flag is not working on Windows. And I could not find a replacement of the Linux/Mac insecure setting for curl

Comment (jrodewig) [reply docs/reference/setup/install/docker.asciidoc:176]: If you're using Powershell, there's a `Invoke-WebRequest` alias for `curl`. You'll need to remove to actually use `curl`:

```
Remove-item alias:curl
```

After that, you can use the `-k` flag.

I'll go ahead and remove this regardless — We probably shouldn't encourage use of the `-k` flag.

**Comment by jrodewig** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.10|[](https://github.com/elastic/elasticsearch/pull/98936)|
|✅|8.9|[](https://github.com/elastic/elasticsearch/pull/98937)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**